sw/device/lib/dif/dif_rv_core_ibex.h

```diff
@@ -159,6 +159,70 @@ typedef enum dif_rv_core_ibex_nmi_source {
   kDifRvCoreIbexNmiSourceAll = 0x3,
 } dif_rv_core_ibex_nmi_source_t;
 
+typedef struct dif_rv_core_ibex_crash_dump_state {
+  /**
+   * The last exception address.
+   */
+  uint32_t mtval;
+
+  /**
+   * The last exception PC.
+   */
+  uint32_t mpec;
+
+  /**
+   * The last data access address.
+   */
+  uint32_t mdaa;
+
+  /**
+   * The next PC.
+   */
+  uint32_t mnpc;
+
+  /**
+   * The current PC.
+   */
+  uint32_t mcpc;
+} dif_rv_core_ibex_crash_dump_state_t;
+
+typedef struct dif_rv_core_ibex_previous_crash_dump_state {
+  /**
+   * The exception address for the previous crash.
+   */
+  uint32_t mtval;
+
+  /**
+   * The last exception PC for the previous crash.
+   */
+  uint32_t mpec;
+} dif_rv_core_ibex_previous_crash_dump_state_t;
+
+/**
+ * Under normal circumstances, only the current crash dump state is valid. When
+ * the CPU encounters a double fault, the current crash dump is moved to
+ * previous, and the new crash dump is shown in current.
+ */
+typedef struct dif_rv_core_ibex_crash_dump_info {
+  /**
+   * The crash dump state for the current execution.
+   */
+  dif_rv_core_ibex_crash_dump_state_t fault_stage;
+
+  /**
+   * The crash dump state for the previous execution. I will only contain valid
+   * data in case of double fault, which will be indicated by the
+   * `double_fault`.
```

Comment:
```suggestion
   * The crash dump state for the previous execution. It will only contain valid
   * data in case of a double fault, which will be indicated by the
   * `double_fault` member.
```

Reply:
done